decoder.go

```diff
@@ -20,18 +42,40 @@ func frameLineSize(frame *C.AVFrame) *C.int {
 	return (*C.int)(unsafe.Pointer(&frame.linesize[0]))
 }
 
-// h264Decoder is a wrapper around ffmpeg's H264 decoder.
-type h264Decoder struct {
-	codecCtx    *C.AVCodecContext
-	srcFrame    *C.AVFrame
-	swsCtx      *C.struct_SwsContext
-	dstFrame    *C.AVFrame
-	dstFramePtr []uint8
+// getStreamInfo opens a stream URL and retrieves its information
+func getStreamInfo(url string) (videoCodec, error) {
+	// Register all formats and codecs
+	C.avformat_network_init()
+
+	cUrl := C.CString(url)
+	defer C.free(unsafe.Pointer(cUrl))
+
+	var avFormatCtx *C.AVFormatContext = nil
+	if C.avformat_open_input(&avFormatCtx, cUrl, nil, nil) != 0 {
```

Comment:
The [docs](https://ffmpeg.org/doxygen/2.8/group__lavf__decoding.html#ga10a404346c646e4ab58f4ed798baca32) say that when there is an error the return value will be an [AVERROR](https://ffmpeg.org/doxygen/trunk/group__lavu__error.html#gae4bb6f165973d09584e0ec0f335f69ca)

Would it be easy to include the error (converted into a go string) in the returned error for all failed calls to libav?

If not, could we maybe at least add the error code as an int in the error message?

Reply:
Added `avError` helper function in Go, that will fetch error message for a given AVERROR and convert it to a gostring.